src/ui/link/index.tsx

```diff
@@ -59,6 +99,20 @@ const Link = forwardRef<HTMLButtonElement | HTMLAnchorElement, LinkProps>(
         to: route && [route, { params, query }],
       });
 
+    const handleMouseEnter = (e: MouseEvent) => {
+      if (prefetch === 'hover') {
+        prefetchRef.current = setTimeout(triggerPrefetch, PREFETCH_DELAY);
```

Comment:
/nit The ref can also be cleared within the `setTimeout` alongside the existing `onMouseLeave` handler